source/store/actions/dynamicFormData.ts

```diff
@@ -0,0 +1,106 @@
+import { Form } from '../../types/FormTypes';
+import { Case } from '../../types/CaseType';
+import { User } from '../../types/UserTypes';
+
+const addIds = (descriptor: string[], questionId: string, formId: string) =>
+  descriptor.map(s => {
+    if (s.split('.')[0] !== 'user' && questionId !== '') {
+      return `${formId}.${questionId}.${s}`;
+    }
+    if (s.split('.')[0] !== 'user' && questionId === '') {
+      return `${formId}.${s}`;
+    }
+    return s;
+  });
+
+/** Takes a form and generates the dataMap template used to fill the dynamical data.  */
+const makeDataMap = (form: Form) => {
```

Comment:
This function name could also be more descriptive, ie genereteFormDataMapTemplate.